DX-1458 TT-12407 update on use of MDCB secret endpoints (#4897)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md

In order to refer to a Tyk MDCB license through a Kubernetes secret, please use `tyk-mdcb.mdcb.useSecretName`, where the secret should contain a key called `MDCBLicense`.

***MDCB Secret***

> In order to set the secret for accessing MDCB secure HTTP endpoints through a Kubernetes secret, please use tyk-mdcb.mdcb.useSecretName, where the secret should contain a key called `securitySecret`.

Reviewer: Add an example of using the secret

> In order to set the secret for accessing MDCB secure HTTP endpoints through a Kubernetes secret, please use tyk-mdcb.mdcb.useSecretName, where the secret should contain a key called `securitySecret`.

Reviewer: ```suggestion
In order to set the secret for accessing MDCB secure HTTP endpoints through a Kubernetes secret, please use tyk-`mdcb.mdcb.useSecretName`, where the secret should contain a key called `securitySecret`.
```

In order to refer to a Tyk MDCB license through a Kubernetes secret, please use `tyk-mdcb.mdcb.useSecretName`, where the secret should contain a key called `MDCBLicense`.

***MDCB Secret***

> In order to set the secret for accessing MDCB secure HTTP endpoints through a Kubernetes secret, please use tyk-mdcb.mdcb.useSecretName, where the secret should contain a key called `securitySecret`.

Author: ```suggestion

In order to set the secret for accessing MDCB secure HTTP endpoints through a Kubernetes secret, please use tyk-mdcb.mdcb.useSecretName, where the secret should contain a key called `securitySecret`.
```

Setting `tyk-mdcb.mdcb.listenPort` field opens a port on MDCB container and MDCB service targets this port.
It is used to set `TYK_MDCB_LISTENPORT`

#### Tyk MDCB HTTP Port

> The HTTP port for Tyk MDCB can be configurable via the `tyk-mdcb.mdcb.httpPort` field.

Reviewer 2: ```suggestion
The HTTP port for Tyk MDCB is configurable via the `tyk-mdcb.mdcb.httpPort` field.
```

> This port lets MDCB allow standard HTTP request like health check.

Reviewer 2: ```suggestion
This port enables MDCB to accept standard HTTP requests, such as health checks.
```

Setting `tyk-mdcb.mdcb.listenPort` field opens a port on MDCB container and MDCB service targets this port.
It is used to set `TYK_MDCB_LISTENPORT`

#### Tyk MDCB HTTP Port

> The HTTP port for Tyk MDCB is configurable via the `tyk-mdcb.mdcb.httpPort` field.

Reviewer: add link to MDCB config field, merged to main branch ready for release

This port enables MDCB to accept standard HTTP requests, such as health checks.

It also defines the path for liveness and readiness probes.
It is used to set `TYK_MDCB_HTTPPORT` in MDCB 2.6.0+ or `TYK_MDCB_HEALTHCHECKPORT` in MDCB 2.5.x or prior.

#### Enabling secured HTTP endpoints

> MDCB has secured HTTP endpoints, like `/dataplanes` which return list of gateway nodes connected. By default, this endpoint is disabled to avoid unintended leakage of data plane information. To enable this endpoint, set `tyk-mdcb.mdcb.security.enableHttpSecureEndpoints` to `true`. It is used to set `TYK_MDCB_SECURITY_ENABLEHTTPSECUREENDPOINTS`. Also, you need to set a secret that can be used to access this endpoint via `tyk-mdcb.mdcb.security.secret` field.

Reviewer: Add link to MDCB OAS API, merged to main ready for release

      volumeName: "mdcb-tls-secret-volume"
```

To enable TLS for MDCB HTTP endpoint, use `tyk-mdcb.mdcb.httpServerOptions` to configure SSL of MDCB HTTP server. The configuration includes settings such as useSSL, certificateKeyFile, certificateCertFile, and minVersion. For other HTTP server options, users can utilize extraEnvs to configure additional parameters.

```yaml
tyk-mdcb:
  mdcb:
    # defines the SSL/TLS settings for the http server where the healthcheck is exposed
    httpServerOptions:
      # if enabled then the endpoints will be served over https

>       useSSL: false

Reviewer: Should this be enabled so that relevant to the example for SSL and TLS and then uncomment the related fields...

Author: good call, docs updated

      # certificateKeyFile: /path-to-cert-keyfile
      # certificateCertFile: /path-to-certfile
      
      # For TLS 1.0 use 769, for TLS 1.1 use 770, for TLS 1.2 use 771, for TLS 1.3 use 772

>       # minVersion: 771

Reviewer: Is there a default is TLS is enabled?

Author: user needs to set it
